crates/metaslang/cst/src/cst.rs

```diff
@@ -62,21 +90,43 @@ impl<T: KindTypes> Node<T> {
         Self::Token(Rc::new(TerminalNode { kind, text }))
     }
 
+    pub fn id(&self) -> usize {
```

Comment:
We probably should add a comment stating that it's supposed to be an opaque identifier that is not guaranteed to be stable across different versions; I know we were discussing potentially using an abstract `NodeId` here rather than the pointer we have now.

Reply:
I'm not sure about the persistence semantics of this id. Should it be the same if you parse the same source twice for example?

Reply:
It shouldn't, but it's important not to make any assumption about the values - now we'll have 0x80001256189, at some point we might switch to lower IDs like 1,2,3 so I think it's worth clarifying this intent to not give any funny idea for the caller to bitpack it themselves at some point etc.

Reply:
I added a comment